continuous_integration/environment-3.10.yaml

```
@@ -0,0 +1,79 @@
+# This job includes coverage
+name: test-environment
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+  # required dependencies
+  - python=3.10
+  - setuptools<60.0
+  - packaging
+  - numpy
+  - pandas
+  # test dependencies
+  - pre-commit
+  - pytest
+  - pytest-cov
+  - pytest-rerunfailures
+  - pytest-xdist
+  - moto
+  - flask
+  # Avoid `SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats`
+  # from thrift on Python 3.10
+  # https://github.com/dask/fastparquet/issues/514
+  # https://github.com/apache/thrift/pull/2487
+  # - fastparquet
+  - h5py
+  - pytables
+  - zarr
+  - tiledb-py
+  # resolver was pulling in old versions, so hard-coding floor
+  # https://github.com/dask/dask/pull/8505
+  - tiledb>=2.5.0
+  - xarray
+  - fsspec
+  # sqlalchemy 1.4.0 causes deprecation warnings to be raised from pandas
+  # along with other issues https://github.com/pandas-dev/pandas/issues/40467
+  # sqlalchemy<1.4.0 doesn't support Python 3.10
+  # - sqlalchemy<1.4.0
```

Elsewhere we have fixed this by moving to SQLAlchemy 1.4+ ( https://github.com/dask/dask/pull/8158 ). Guessing this is just an outdated copy and we can do the same here

```suggestion
  - sqlalchemy>=1.4.0
```